src/components/Recommendation.tsx

```diff
@@ -0,0 +1,164 @@
+
+const periodNumbers = [
+  '2022/2',
+  '2023/1',
+  '2023/2',
+  '2024/1',
+  '2024/2',
+  '2025/1',
+  '2025/2',
+  '2026/1',
+  '2026/2',
+  '2027/1',
+  '2027/2',
+  '2028/1',
+  '2028/2',
+  '2029/1',
+  '2029/2',
+  '2030/1',
+  '2030/2',
+]
+
+const OverlayWrapper = ({ children }: { children: ReactNode }) => (
+  <Box
+    position="absolute"
+    sx={{ background: '#f5f5f5', inset: '0 0 0 0' }}
+    display="flex"
+    alignItems="center"
+    justifyContent="center"
+  >
+    {children}
+  </Box>
+)
+
+const RecommendationLoading = () => (
+  <OverlayWrapper>
+    <CircularProgress color="secondary" size="2rem" thickness={5} />
+  </OverlayWrapper>
+)
+
+export default function RecommendationContainer() {
+  const recommendation: Recommendation | null =
+    useSelector(selectRecommendation)
+  const isLoading = recommendation === null
+
+  return (
+    <Grid container spacing={2} padding={0} paddingLeft={0}>
+      {isLoading && <RecommendationLoading />}
+      {recommendation && (
+        <>
+          <Grid item xs={12}>
+            <Typography variant="body1">
+              Cursando até{' '}
+              <strong> {recommendation.maxCreditsByPeriod} créditos</strong> em
+              cada semestre, serão necessários
+              <strong> {recommendation.periods.length} semestres </strong>
+              para você se formar
+            </Typography>
+          </Grid>
+          <Grid item xs={12} padding={0} marginTop="8px">
+            {recommendation.periods.map((period: Period, index) => {
+              return (
+                <Grid container padding={0}>
+                  <Grid item xs={1} display="flex" alignItems="center">
+                    <Box
+                      sx={{
+                        backgroundColor: 'primary.main',
+                        color: 'primary.contrastText',
+                        height: '70px',
+                        width: '70px',
+                        textAlign: 'center',
+                        borderRadius: '50%',
+                        display: 'flex',
+                        justifyContent: 'center',
+                        alignItems: 'center',
+                      }}
+                    >
+                      <Typography variant="body1">
+                        <strong>{periodNumbers[index]}</strong>
+                      </Typography>
+                    </Box>
+                  </Grid>
+                  <Grid item xs={11} container paddingLeft="6px">
+                    <Box
+                      display="flex"
+                      flexWrap="wrap"
+                      padding={0}
+                      margin={0}
+                      marginY="16px"
+                      width="100%"
+                      justifyContent="center"
+                    >
+                      {period.courses.map((course: Course) => {
+                        return (
+                          <Box
```

Comment:
Acho que é aqui que tá faltando a prop `key`
```suggestion
                          <Box 
                            key={course.id}
```

![image](https://user-images.githubusercontent.com/37981839/189254302-34fd9d0b-121a-495b-92f9-9739fdfa0167.png)